swagger generate spec panics with "invalid memory address or nil pointer dereference" on Go 1.24.6

I am experiencing a crash when running swagger generate spec on my project. It consistently panics with a nil pointer dereference, and I cannot upgrade Swagger at the moment.

```
[root@vralabs1121 peerserver]# swagger generate spec -w . -o /tmp/peerserver.json
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xafacbe]

goroutine 165 [running]:
go/types.(*Checker).handleBailout(0xc0006a4000, 0xc000a3fc90)
        /usr/lib/golang/src/go/types/check.go:434 +0x88
panic({0xc98820?, 0x1603700?})
        /usr/lib/golang/src/runtime/panic.go:792 +0x132
go/types.(*StdSizes).Sizeof(0x0, {0xedd9c8, 0x160e840})
        /usr/lib/golang/src/go/types/sizes.go:229 +0x31e
go/types.(*Config).sizeof(...)
        /usr/lib/golang/src/go/types/sizes.go:334
go/types.representableConst.func1(...)
        /usr/lib/golang/src/go/types/const.go:77
go/types.representableConst({0xee1e70, 0xf03418}, 0xc0006a4000, 0x160e840, 0xc000a3ef80)
        /usr/lib/golang/src/go/types/const.go:107 +0x387
go/types.(*Checker).representation(0xc0006a4000, 0xc00039bb00, 0x160e840)
        /usr/lib/golang/src/go/types/const.go:257 +0x5f
go/types.(*Checker).representable(0xc0006a4000, 0xc00039bb00, 0x160e840)
        /usr/lib/golang/src/go/types/const.go:240 +0x26
go/types.(*Checker).shift(0xc0006a4000, 0xc00039ba80, 0xc00039bb00, {0xee0b80, 0xc00089d650}, 0x14)
        /usr/lib/golang/src/go/types/expr.go:627 +0x1eb
go/types.(*Checker).binary(0xc0006a4000, 0xc00039ba80, {0xee0b80, 0xc00089d650}, {0xee07c0, 0xc000a282a0}, {0xee07c0, 0xc000a282c0}, 0x14, 0x3882)
        /usr/lib/golang/src/go/types/expr.go:769 +0x150
go/types.(*Checker).exprInternal(0xc0006a4000, 0x0, 0xc00039ba80, {0xee0b80, 0xc00089d650}, {0x0?, 0x0?})
        /usr/lib/golang/src/go/types/expr.go:1130 +0x6be
go/types.(*Checker).rawExpr(0xc0006a4000, 0x0, 0xc00039ba80, {0xee0b80?, 0xc00089d650?}, {0x0?, 0x0?}, 0x0)
        /usr/lib/golang/src/go/types/expr.go:952 +0x18c
go/types.(*Checker).expr(0xc0006a4000, 0xc0006da670?, 0xc00039ba80, {0xee0b80?, 0xc00089d650?})
        /usr/lib/golang/src/go/types/expr.go:1227 +0x30
go/types.(*Checker).binary(0xc0006a4000, 0xc00039ba80, {0xee0b80, 0xc00089d680}, {0xee0b80, 0xc00089d650}, {0xee07c0, 0xc000a282e0}, 0xd, 0x3886)
        /usr/lib/golang/src/go/types/expr.go:756 +0xa5
go/types.(*Checker).exprInternal(0xc0006a4000, 0x0, 0xc00039ba80, {0xee0b80, 0xc00089d680}, {0x0?, 0x0?})
        /usr/lib/golang/src/go/types/expr.go:1130 +0x6be
go/types.(*Checker).rawExpr(0xc0006a4000, 0x0, 0xc00039ba80, {0xee0b80?, 0xc00089d680?}, {0x0?, 0x0?}, 0x0)
        /usr/lib/golang/src/go/types/expr.go:952 +0x18c
go/types.(*Checker).expr(0xc0006a4000, 0xc000a7b800?, 0xc00039ba80, {0xee0b80?, 0xc00089d680?})
        /usr/lib/golang/src/go/types/expr.go:1227 +0x30
go/types.(*Checker).constDecl(0xc0006a4000, 0xc000a7b860, {0x0, 0x0}, {0xee0b80, 0xc00089d680}, 0x0)
        /usr/lib/golang/src/go/types/decl.go:473 +0x2b8
go/types.(*Checker).objDecl(0xc0006a4000, {0xee92f8, 0xc000a7b860}, 0x0)
        /usr/lib/golang/src/go/types/decl.go:184 +0x98c
go/types.(*Checker).packageObjects(0xc0006a4000)
        /usr/lib/golang/src/go/types/resolver.go:690 +0x471
go/types.(*Checker).checkFiles(0xc0006a4000, {0xc000424010?, 0xabbb05?, 0xd4eba0?})
        /usr/lib/golang/src/go/types/check.go:489 +0x2a6
go/types.(*Checker).Files(0xc00017c1c0?, {0xc000424010?, 0xc000a16120?, 0x4?})
        /usr/lib/golang/src/go/types/check.go:452 +0x75
golang.org/x/tools/go/packages.(*loader).loadPackage(0xc00017c1c0, 0xc0006f7640)
        /root/go/pkg/mod/golang.org/x/tools@v0.1.0/go/packages/packages.go:955 +0x70f
golang.org/x/tools/go/packages.(*loader).loadRecursive.func1()
        /root/go/pkg/mod/golang.org/x/tools@v0.1.0/go/packages/packages.go:793 +0x1a9
sync.(*Once).doSlow(0x0?, 0x0?)
        /usr/lib/golang/src/sync/once.go:78 +0xab
sync.(*Once).Do(...)
        /usr/lib/golang/src/sync/once.go:69
golang.org/x/tools/go/packages.(*loader).loadRecursive(0x0?, 0x0?)
        /root/go/pkg/mod/golang.org/x/tools@v0.1.0/go/packages/packages.go:781 +0x3b
golang.org/x/tools/go/packages.(*loader).loadRecursive.func1.1(0x0?)
        /root/go/pkg/mod/golang.org/x/tools@v0.1.0/go/packages/packages.go:788 +0x26
created by golang.org/x/tools/go/packages.(*loader).loadRecursive.func1 in goroutine 139
        /root/go/pkg/mod/golang.org/x/tools@v0.1.0/go/packages/packages.go:787 +0x94
[root@vralabs1121 peerserver]# 

```
[root@vralabs1121 peerserver]# go version
go version go1.24.6 (Red Hat 1.24.6-1.module+el8.10.0+23407+428597c7) linux/amd64
[root@vralabs1121 peerserver]# swagger version
version: v0.27.0
commit: (unknown, mod sum: "h1:K7+nkBuf4oS1jTBrdvWqYFpqD69V5CN8HamZzCDDhAI=")
[root@vralabs1121 peerserver]#

[root@vralabs1121 peer]# make docs
mkdir -p /workspace/thunder/services/peer/build_dir/docs \
    && export SWAGGER_GENERATE_EXTENSION=false \
    && swagger generate spec -c flex/services/common/src/error -c ./api/src/peerserver -w ./api/src/peerserver/ -o /workspace/thunder/services/peer/build_dir/docs/peerserver-restapis.json \
    && cp ./README.md / \
    && cp /workspace/thunder/services/peer/build_dir/docs/peerserver-restapis.json /swagger/
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xafacbe]

goroutine 21 [running]:
go/types.(*Checker).handleBailout(0xc0001963c0, 0xc0007ffc90)
        /usr/lib/golang/src/go/types/check.go:434 +0x88
panic({0xc98820?, 0x1603700?})
        /usr/lib/golang/src/runtime/panic.go:792 +0x132
go/types.(*StdSizes).Sizeof(0x0, {0xedd9c8, 0x160e7a0})
        /usr/lib/golang/src/go/types/sizes.go:229 +0x31e
go/types.(*Config).sizeof(...)
        /usr/lib/golang/src/go/types/sizes.go:334
go/types.representableConst.func1(...)
        /usr/lib/golang/src/go/types/const.go:77
go/types.representableConst({0xee1e70, 0xf033e0}, 0xc0001963c0, 0x160e7a0, 0xc0007fe910)
        /usr/lib/golang/src/go/types/const.go:93 +0x1e9
go/types.(*Checker).representation(0xc0001963c0, 0xc000692900, 0x160e7a0)
        /usr/lib/golang/src/go/types/const.go:257 +0x5f
go/types.(*Checker).implicitTypeAndValue(0xc0001963c0, 0xc000692900, {0xedd9c8, 0x160e7a0})
        /usr/lib/golang/src/go/types/expr.go:374 +0x6d9
go/types.(*Checker).convertUntyped(0xc0001963c0, 0xc000692900, {0xedd9c8, 0x160e7a0})
        /usr/lib/golang/src/go/types/const.go:290 +0x3f
go/types.(*Checker).matchTypes(0xc0001963c0, 0xc000692480, 0xc000692900)
        /usr/lib/golang/src/go/types/expr.go:899 +0x315
go/types.(*Checker).binary(0xc0001963c0, 0xc000692480, {0xee0b80, 0xc00088e7b0}, {0xee10c0, 0xc00089f080}, {0xee07c0, 0xc000891660}, 0x27, 0x12f5)
        /usr/lib/golang/src/go/types/expr.go:773 +0x166
go/types.(*Checker).exprInternal(0xc0001963c0, 0x0, 0xc000692480, {0xee0b80, 0xc00088e7b0}, {0x0?, 0x0?})
        /usr/lib/golang/src/go/types/expr.go:1130 +0x6be
go/types.(*Checker).rawExpr(0xc0001963c0, 0x0, 0xc000692480, {0xee0b80?, 0xc00088e7b0?}, {0x0?, 0x0?}, 0x0)
        /usr/lib/golang/src/go/types/expr.go:952 +0x18c
go/types.(*Checker).expr(0xc0001963c0, 0x4e8899e3f709d93d?, 0xc000692480, {0xee0b80?, 0xc00088e7b0?})
        /usr/lib/golang/src/go/types/expr.go:1227 +0x30
go/types.(*Checker).binary(0xc0001963c0, 0xc000692440, {0xee0b80, 0xc00088e7e0}, {0xee0b80, 0xc00088e780}, {0xee0b80, 0xc00088e7b0}, 0x22, 0x12e4)
        /usr/lib/golang/src/go/types/expr.go:757 +0xcc
go/types.(*Checker).exprInternal(0xc0001963c0, 0x0, 0xc000692440, {0xee0b80, 0xc00088e7e0}, {0x0?, 0x0?})
        /usr/lib/golang/src/go/types/expr.go:1130 +0x6be
go/types.(*Checker).rawExpr(0xc0001963c0, 0x0, 0xc000692440, {0xee0b80?, 0xc00088e7e0?}, {0x0?, 0x0?}, 0x0)
        /usr/lib/golang/src/go/types/expr.go:952 +0x18c
go/types.(*Checker).expr(0xc0001963c0, 0x160e780?, 0xc000692440, {0xee0b80?, 0xc00088e7e0?})
        /usr/lib/golang/src/go/types/expr.go:1227 +0x30
go/types.(*Checker).initVars(0xc0001963c0, {0xc0007f0508, 0x1, 0xc0008913a0?}, {0xc000882440, 0xedd900?, 0xedd900?}, {0xee0df0, 0xc000891680})
        /usr/lib/golang/src/go/types/assignments.go:412 +0x608
go/types.(*Checker).stmt(0xc0001963c0, 0x0, {0xee0df0, 0xc000891680})
        /usr/lib/golang/src/go/types/stmt.go:575 +0x685
go/types.(*Checker).stmtList(0xc0001963c0, 0x0, {0xc000882450?, 0x1?, 0xc0001963c0?})
        /usr/lib/golang/src/go/types/stmt.go:123 +0x85
go/types.(*Checker).funcBody(0xc0001963c0, 0xee0670?, {0xc0008802b0?, 0xc000891360?}, 0xc000692380, 0xc00088e810, {0x0?, 0x0?})
        /usr/lib/golang/src/go/types/stmt.go:43 +0x311
go/types.(*Checker).funcDecl.func1()
        /usr/lib/golang/src/go/types/decl.go:868 +0x3a
go/types.(*Checker).processDelayed(0xc0001963c0, 0x0)
        /usr/lib/golang/src/go/types/check.go:550 +0x1e4
go/types.(*Checker).checkFiles(0xc0001963c0, {0xc000066af0?, 0xabbb05?, 0xd4eba0?})
        /usr/lib/golang/src/go/types/check.go:492 +0x336
go/types.(*Checker).Files(0xc000000000?, {0xc000066af0?, 0xc0009b4060?, 0x5?})
        /usr/lib/golang/src/go/types/check.go:452 +0x75
golang.org/x/tools/go/packages.(*loader).loadPackage(0xc000000000, 0xc0006d67e0)
        /root/go/pkg/mod/golang.org/x/tools@v0.1.0/go/packages/packages.go:955 +0x70f
golang.org/x/tools/go/packages.(*loader).loadRecursive.func1()
        /root/go/pkg/mod/golang.org/x/tools@v0.1.0/go/packages/packages.go:793 +0x1a9
sync.(*Once).doSlow(0x0?, 0x0?)
        /usr/lib/golang/src/sync/once.go:78 +0xab
sync.(*Once).Do(...)
        /usr/lib/golang/src/sync/once.go:69
golang.org/x/tools/go/packages.(*loader).loadRecursive(0x0?, 0x0?)
        /root/go/pkg/mod/golang.org/x/tools@v0.1.0/go/packages/packages.go:781 +0x3b
golang.org/x/tools/go/packages.(*loader).refine.func2(0x0?)
        /root/go/pkg/mod/golang.org/x/tools@v0.1.0/go/packages/packages.go:722 +0x26
created by golang.org/x/tools/go/packages.(*loader).refine in goroutine 1
        /root/go/pkg/mod/golang.org/x/tools@v0.1.0/go/packages/packages.go:721 +0xbb9
make: *** [Makefile:101: docs] Error 2

--- a/codescan/schema.go
+++ b/codescan/schema.go
@@ -1692,7 +1692,11 @@ func isTextMarshaler(tpe types.Type) bool {
 	if err != nil {
 		return false
 	}
-	ifc := encodingPkg.Scope().Lookup("TextMarshaler").Type().Underlying().(*types.Interface) // TODO: there is a better way to check this
+	obj := encodingPkg.Scope().Lookup("TextMarshaler")
+	if obj == nil {
+		return false
+	}
+	ifc := obj.Type().Underlying().(*types.Interface) // TODO: there is a better way to check this
 
 	return types.Implements(tpe, ifc)
 }

